src/crimson/os/seastore/cache.cc

```diff
@@ -1661,7 +1666,12 @@ Cache::replay_delta(
       delta.paddr != P_ADDR_NULL &&
       delta.paddr.get_addr_type() == paddr_types_t::SEGMENT) {
     auto& seg_addr = delta.paddr.as_seg_paddr();
-    auto& seg_info = segment_provider->get_seg_info(seg_addr.get_segment_id());
+    auto provider = segment_provider;
+    if (cold_segment_provider &&
+	!segment_provider->contains(seg_addr.get_segment_id())) {
+      provider = cold_segment_provider;
```

Comment:
```c++
ceph_assert(provider->contains(sid));
```

Reply:
The comment is dismissed, see https://github.com/ceph/ceph/pull/47974#discussion_r1061095999